src/main/java/net/dries007/tfc/common/blocks/WeatheringMetalBlock.java

```diff
@@ -0,0 +1,94 @@
+package net.dries007.tfc.common.blocks;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.BlockHitResult;
+
+import net.dries007.tfc.common.TFCTags;
+import net.dries007.tfc.util.Metal;
+import net.dries007.tfc.util.registry.RegistryMetal;
+
+public class WeatheringMetalBlock extends Block implements IClimateWeatheringBlock
+{
+    static final float BASE_METAL_CHANCE = 0.015f;
+    TFCWeatherState weatherState;
+    RegistryMetal metal;
+
+    public WeatheringMetalBlock(Properties properties, TFCWeatherState weatherState, RegistryMetal metal)
+    {
+        super(properties);
+        this.weatherState = weatherState;
+        this.metal = metal;
+    }
+
+    @Override
+    public void randomTick(BlockState state, ServerLevel serverLevel, BlockPos pos, RandomSource random) {
+        if (random.nextFloat() < BASE_METAL_CHANCE) {
```

Comment:
This chance check should be in `IClimateWeatheringBlock`, since it's common in all our implementations